### PR TITLE
TINY-6887: Change the action order

### PR DESCRIPTION
**Pull request by ArvinJ-H**
Related Ticket: 
TINY-6887

Description of Changes:
- Changed the order actions, sandbox is closed before action is triggered

Pre-checks:
* [x] Changelog entry added
* [x] <s>Tests have been added (if applicable)</s>
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

**Review by spocke — COMMENTED** [review]
Can we add tests for it so it doesn't come back?

**Comment by TheSpyder** [comment]
If `onAction` causes the menu to close, do we even need the close sandbox line at all?

**Comment by TheSpyder** [comment]
> If `onAction` causes the menu to close, do we even need the close sandbox line at all?

After digging in a bit, the answer is yes. Why the menu closes is an interesting question. As near as I can tell, it's because `spec.onAction` is this `editor.execCommand` call, and exec command will focus the editor by default. The change in focus closes the menu.
https://github.com/tinymce/tinymce/blob/f6a87ed7a6d32c627cf8ac6c1bd44ce36d69b14b/modules/tinymce/src/plugins/table/main/ts/ui/MenuItems.ts#L61-L64

However this only happens if the table plugin is active. In practice the table grid picker won't be rendered without the table plugin, but in theory it could be used for something else.

**Review by TheSpyder — APPROVED** [review]

File: modules/tinymce/src/plugins/table/test/ts/browser/TableContextMenuInsertTest.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    const warnSanitized = sanitizeString(lastWarn);
+    const expectedSanitized = sanitizeString(expectedLog);
+
+    if (warnSanitized === expectedSanitized) {
+      throw new Error(`Warning matched expected: ${expectedSanitized}`);
+    }
+  };
+
+  it('TINY-6887', async () => {
```
Comment (shanmen-tiny) [inline modules/tinymce/src/plugins/table/test/ts/browser/TableContextMenuInsertTest.ts:42]: Description

File: modules/tinymce/src/plugins/table/test/ts/browser/TableContextMenuInsertTest.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    console.warn = originalWarn;
+
+    const lastWarn = Arr.last(warns).getOr('none');
+
+    const warnSanitized = sanitizeString(lastWarn);
+    const expectedSanitized = sanitizeString(expectedLog);
+
+    if (warnSanitized === expectedSanitized) {
+      throw new Error(`Warning matched expected: ${expectedSanitized}`);
+    }
```
Comment (shanmen-tiny) [inline modules/tinymce/src/plugins/table/test/ts/browser/TableContextMenuInsertTest.ts:39]:
```suggestion
    assert.notEqual(warnSanitized, expectedSanitized, `Warning matched expected: ${expectedSanitized}`);
```

File: modules/tinymce/src/plugins/table/test/ts/browser/TableContextMenuInsertTest.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  };
+
+  it('TINY-6887', async () => {
+    const editor = hook.editor();
+    editor.setContent('<p></p>');
+    await pOpenContextMenu(editor, 'p');
+    await TinyUiActions.pWaitForPopup(editor, '.tox-collection--list');
+    TinyUiActions.clickOnUi(editor, 'div[role="menuitem"]:contains("Table")');
+    await assertDebugLog( editor, 'The component must be in a context to execute: triggerEvent <div class="tox-insert-table-picker"></div> is not in context.');
```
Comment (shanmen-tiny) [inline modules/tinymce/src/plugins/table/test/ts/browser/TableContextMenuInsertTest.ts:48]: Maybe rename to `pAssertNoConsoleWarn` or something and move the second parameter into the function? This looks like it's expecting the string but it's throwing error in that function when it matches 🤪

**Review by shanmen-tiny — APPROVED** [review]

File: modules/tinymce/src/plugins/table/test/ts/browser/TableContextMenuInsertTest.ts (hunk 3 above)
Comment (ArvinJ-H) [reply modules/tinymce/src/plugins/table/test/ts/browser/TableContextMenuInsertTest.ts:46]: oh yeah that function name needs to be updated, i'll move the sting into the function

**Pull request merged**